### DISCUSSION TITLE
[Vite + SSR] Is it possible to build to ES instead of CJS?

### DISCUSSION THREAD

**Discussion by gustavotoyota** [quasarframework/quasar#14733] [CLI - SSR mode]
I'm trying to use some newer functionality like top-level await on the server.
I managed to build the server to ES doing this:
```
extendViteConf(viteConf, invokeParams) {
  if (invokeParams.isServer) {
    viteConf.build.rollupOptions.output = { format: 'es' };
  }
},
```
But the file `dist/ssr/index.js` is still being built as CJS, so I'm unable to start the server.

Is it possible to build everything into ES code?

**Comment by stefanvanherwijnen** [comment]
The index.js file seems to be build with esbuild: https://github.com/quasarframework/quasar/blob/8f0ccef64ca3b6a44ed459d5b0afef02a17801c5/app-vite/lib/modes/ssr/ssr-builder.js#L70
Not sure if you can change the configuration.

**Reply by gustavotoyota** [reply]
I managed to build the index.js to ES, but it isn't very compatible with the rest of the code.
It seems like Quasar would need a good rewrite to be able to build to ES code.
